### ISSUE TITLE
total_num_steps for scheduler

### ISSUE THREAD

**Issue by SihengLi99** [huggingface/accelerate#2318]
### System Info

```Shell
accelerate 0.25
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

<img width="861" alt="image" src="https://github.com/huggingface/accelerate/assets/75015153/0ae37100-3150-42ba-98c5-8fa1f2ff1319">

### Expected behavior

<img width="861" alt="image" src="https://github.com/huggingface/accelerate/assets/75015153/0ae37100-3150-42ba-98c5-8fa1f2ff1319">
Why divide the num_processes when preparing the lr_scheduler?
It seems strange because I have computed the true total_num_steps before accelerator.prepare().

**Comment by SunMarc** [comment]
cc @pacman100

**Comment by muellerzr** [comment]
By default we assume a user does not take the effort to compute the true number of steps, since the code is designed to work on any number of machines. If you are doing this, you can pass `step_with_optimizer=False` to the `Accelerator` so it will only step once when you want to.
